R/cluster_sce.R

```diff
@@ -0,0 +1,27 @@
+#' Perform clustering on a SingleCellExperiment object
+#'
+#' @param sce SingleCellExperiment object that requires clustering
+#' @param pca_name The name used to access the PCA results
+#' @param BLUSPARAM A BlusterParam object specifying the clustering algorithm to
+#' use - can be `KmeansParam` or `NNGraphParam`
+#' @param cluster_name The type of clustering method to be performed - can be
+#' "kmeans", "walktrap", or "louvain"
+#' @param seed Seed for reproducibility of clustering results
+#'
+#' @return SingleCellExperiment object containing clustering results
+cluster_sce <- function(sce,
+                        pca_name,
+                        BLUSPARAM,
+                        cluster_name,
+                        seed = NULL,
+                        ...) {
+  # Set the seed
+  set.seed(seed)
+
```

Comment:
You will also need to add in checks here that the pcs exist in the SCE object and then we will probably want to check the `BLUSPARAM` object too. I'm not quite sure how we would do that though...

Reply:
I would probably leave the `BLUSPARAM` checks to `bluster` in this case.